### PR TITLE
Issue fix 5043

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Fix issue #5043, after the restore pod is scheduled, check if the node-agent pod is running in the same node.

### CODE REVIEW

File: pkg/podvolume/restorer.go [hunk 1]
```diff
@@ -129,6 +145,8 @@ func (r *restorer) RestorePodVolumes(data RestoreData) []error {
 	r.results[resultsKey(data.Pod.Namespace, data.Pod.Name)] = resultsChan
 	r.resultsLock.Unlock()
 
+	r.nodeAgentCheck = make(chan struct{})
```
Comment (reasonerjt) [inline pkg/podvolume/restorer.go:148]: how about making it an error channel so we have more information of the node name in the error message?

File: pkg/util/kube/pod.go [hunk 2]
```diff
@@ -37,3 +37,21 @@ func IsPodRunning(pod *corev1api.Pod) error {
 
 	return nil
 }
+
+// IsPodRunning does a well-rounded check to make sure the specified pod has been scheduled into a node and in a stable status.
+// If not, return the error found
+func IsPodScheduled(pod *corev1api.Pod) error {
```
Comment (reasonerjt) [inline pkg/util/kube/pod.go:43]: Could you extract the dup code of `IsPodRunning` and `IsPodRunningOnNode` into a shared package private func?

File: pkg/util/kube/pod.go [hunk 3]
```diff
@@ -37,3 +37,21 @@ func IsPodRunning(pod *corev1api.Pod) error {
 
 	return nil
 }
+
+// IsPodRunning does a well-rounded check to make sure the specified pod has been scheduled into a node and in a stable status.
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/util/kube/pod.go]: Done

File: pkg/util/kube/pod.go (hunk 2 above)
Comment (Lyndon-Li) [reply pkg/util/kube/pod.go:37]: Done

**Review by reasonerjt — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

File: pkg/util/kube/pod.go [hunk 4]
```diff
@@ -23,12 +23,38 @@ import (
...
+		return nil
+	})
+}
+
+// IsPodScheduled does a well-rounded check to make sure the specified pod has been scheduled into a node and in a stable status.
+// If not, return the error found
+func IsPodScheduled(pod *corev1api.Pod) error {
+	return isPodScheduledInStatus(pod, func(pod *corev1api.Pod) error {
+		if pod.Status.Phase != corev1api.PodRunning && pod.Status.Phase != corev1api.PodPending {
+			return errors.New("pod is running or pending")
```
[the remark this answers is not in the document]
Comment (Lyndon-Li) [reply pkg/util/kube/pod.go:40]: Yes, thanks. Let me merge this PR, I will correct it in other PRs.

**Pull request merged**